src/pages/PeoplePage/PeoplePage.tsx

```diff
@@ -1,7 +1,65 @@
-import React, { FC } from "react";
+import React, { FC, useEffect } from "react";
+import { useAppConfigContext } from "../../app";
+import useFetchData, {
+  FetchDataActionType,
+} from "../../containers/FetchDataContainer/useFetchData";
+import FetchDataContainer from "../../containers/FetchDataContainer/FetchDataContainer";
+import { PeopleContainer } from "../../containers/PeopleContainer";
+import { PeoplePageData } from "../../containers/PeopleContainer/types";
+import { filterRolesByTitle, ROLE_TITLE } from "../../models/util/RoleUtilities";
+import { createError } from "../../utils/createError";
+import RoleService from "../../networking/RoleService";
+import PersonService from "../../networking/PersonService";
 
 const PeoplePage: FC = () => {
-  return <div>TODO</div>;
+  // Get the app config context
+  const { firebaseConfig } = useAppConfigContext();
+
+  // Initialize the state of fetching the people's data
+  const { state: peopleDataState, dispatch: personDataDispatch } = useFetchData<PeoplePageData>({
+    isLoading: false,
+  });
+
+  useEffect(() => {
+    const rolesService = new RoleService(firebaseConfig);
+    const personService = new PersonService(firebaseConfig);
+    let didCancel = false;
+
+    const fetchPersonData = async () => {
+      personDataDispatch({ type: FetchDataActionType.FETCH_INIT });
+
+      try {
+        const currentPeople = await rolesService.getCurrentRoles();
+        const allPeople = await personService.getAllPeople();
+        if (!didCancel) {
+          personDataDispatch({
+            type: FetchDataActionType.FETCH_SUCCESS,
+            payload: {
+              currentPeople: filterRolesByTitle(currentPeople, ROLE_TITLE.COUNCILMEMBER),
+              allPeople,
+            },
+          });
+        }
+      } catch (err) {
+        if (!didCancel) {
+          const error = createError(err);
+          personDataDispatch({ type: FetchDataActionType.FETCH_FAILURE, payload: error });
+        }
+      }
+    };
+
+    fetchPersonData();
+
+    return () => {
+      didCancel = true;
+    };
+  }, []);
```

Comment:
I modified the useFetchData hook a bit in my PR. I just pass the fetching function to the hook and let the hook handle setting the status of fetching -- is loading, has error, and the data itself.

This will eventually need to change to (depending on which PR gets merge first)
```
const fetchPeoplePageData = useCallback( async () => {
    const rolesService = new RoleService(firebaseConfig);
    const personService = new PersonService(firebaseConfig);

    const [currentPeople, allPeople] = await Promise.all([
      rolesService.getCurrentRoles(),
      personService.getAllPeople(),
    ])
    return Promise.resolve({
      allPeople,
      currentPeople: filterRolesByTitle(currentPeople, ROLE_TITLE.COUNCILMEMBER),
    })
  }, [firebaseConfig])

  // Initialize the state of fetching the people's data
  const { state: peopleDataState, dispatch: personDataDispatch } = useFetchData<PeoplePageData>(
    {
      isLoading: false,
      error: null,
      hasFetchRequest: true,
    },
    fetchPeoplePageData
  );
```

Reply:
This will all get resolve with either your PR or the re-do of the PeoplePage, it's really just a placeholder in this PR.

Reply:
My PR has been merged and it doesn't contain the above code. Merging this PR as is will create an error since `useFetchData` has changed.

The change to `useFetchData` is mostly just a refactor. It moves the useEffect(line 23) hook inside `useFetchData` so we don't need to write this useEffect hook for every page. In order to fetch the data inside `useFetchData` the hook now take an additional fetch function argument to fetch the data. 

```
const fetchPeoplePageData = useCallback( async () => {
    const rolesService = new RoleService(firebaseConfig);
    const personService = new PersonService(firebaseConfig);

    const [currentPeople, allPeople] = await Promise.all([
      rolesService.getCurrentRoles(),
      personService.getAllPeople(),
    ])
    return Promise.resolve({
      allPeople,
      currentPeople: filterRolesByTitle(currentPeople, ROLE_TITLE.COUNCILMEMBER),
    })
  }, [firebaseConfig])
```
is the fetch function, but memoized (by wrapping inside a useCallback) so that it doesn't change on every re-render of the page.